pylinalg/obj/transform.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+
+from ..func import quaternion as qt
+
+
+class Transform:
+    """
+    Abstract base class to define a common interface and allow future extension
+    to non-linear transformations.
+    """
+
+
+class AffineTransform(Transform):
+    """
+    A affine transformation.
+
+    This transform represents an affine mapping between two coordinate frames:
+    parent and child. That is, a transformation that is linear in homogeneous
+    coordinates and that consists of a translation, rotation, and scale (applied
+    in that order).
+
+
+    Parameters
+    ----------
+    position : ndarray, [3]
+        The position of the child's frame in the parent's space. (The
+        translation to apply to map vectors from child to parent.)
+    orientation : ndarray, [4]
+        The orientation of the parent's frame relative to the child's frame. (A
+        quaternion (format: x, y, z, w) describing the rotation from child to
+        parent.)
```

Comment:
Isn't it the other way around? When I set a child's rotation to 90 degrees about the y axis, I expect the child to rotate, not the parent.

Reply:
You are correct. I wrote the wrong documentation here. It should indeed be the orientation of the target frame when seen from the parent.